File: src/main/java/org/example/kalkulationsprogramm/dto/Anfrage/AnfrageSeiteResponseDto.java
package org.example.kalkulationsprogramm.dto.Anfrage;

import java.util.List;

/**
 * Antwort für den paginierten Anfragen-Listenendpunkt
 * (analog zu {@code ProjektSearchResponseDto}). Wird ausgeliefert, sobald der
 * Aufrufer einen {@code page}-Query-Parameter setzt.
 */
public record AnfrageSeiteResponseDto(
        List<AnfrageResponseDto> anfragen,
        long gesamt,
        int seite,
        int seitenGroesse) {
}

File: src/main/kotlin/org/example/kalkulationsprogramm/dto/Anfrage/AnfrageSeiteResponseDto.kt
package org.example.kalkulationsprogramm.dto.Anfrage
import java.util.List
/**
 * Antwort für den paginierten Anfragen-Listenendpunkt
 * (analog zu {@code ProjektSearchResponseDto}). Wird ausgeliefert, sobald der
 * Aufrufer einen {@code page}-Query-Parameter setzt.
 */
data class AnfrageSeiteResponseDto(
    val anfragen: List<AnfrageResponseDto>,
    val gesamt: Long,
    val seite: Int,
    val seitenGroesse: Int
) {
    fun anfragen(): List<AnfrageResponseDto> = anfragen
    fun gesamt(): Long = gesamt
    fun seite(): Int = seite
    fun seitenGroesse(): Int = seitenGroesse
}
